lua/neoclip/handlers.lua

```diff
@@ -94,4 +99,24 @@ M.delete = function(typ, entry)
     storage.delete(typ, entry)
 end
 
+M.handle_system_yank = function ()
+    if neoclip.stopped then
+        return
+    end
+    if not settings.enable_system_clipboard then
+        return
+    end
+    for _, register in ipairs({'+', '*'}) do
+        local event = get_register_event(register)
+        if #event.regcontents ~= 0 then
+            if should_add(event) then
```

Comment:
What do you think about extracting a function from the current `handle_text_yank` which takes and `event` table and does the check for `should_add` and `neoclip.stopped` and we don't need to check it in more than one place?